src/Microsoft.OData.Core/ODataBatchOperationRequestMessage.cs

```diff
@@ -92,6 +103,22 @@ public string Method
         }
 
         /// <summary>
+        /// Gets the request prerequisites.
+        /// </summary>
+        public IList<string> DependsOnRequestIds
+        {
+            get
+            {
+                return this.dependsOnRequestIds;
+            }
+
+            set
+            {
+                this.dependsOnRequestIds = value;
+            }

```

Comment:
I wouldn't expose the setter, just make sure it is initialized to a list containing any dependencyIds passed in the constructor, and the app can add new dependency Ids to the list.